jdaviz/configs/default/plugins/export_plot/export_plot.py

```diff
@@ -74,3 +112,184 @@ def vue_save_figure(self, filetype):
+
+        fps : float or `None`
+            Frame rate in frames per second (FPS).
+            If not given, it is obtained from plugin inputs.
+
+        filename : str or `None`
+            Filename for the movie to be recorded. Include path if necessary.
+            If not given, it is obtained from plugin inputs.
+            If another file with the same name already exists, it will be silently replaced.
+
+        filetype : {'mp4', `None`}
+            Currently only MPEG-4 is supported. This keyword is reserved for future support
+            of other format(s).
+
+        rm_temp_files : bool
+            Remove temporary PNG files after movie creation. Default is `True`.
+
+        Returns
+        -------
+        out_filename : str
+            The absolute path to the actual output file.
+
+        """
+        if self.config != "cubeviz":
+            raise NotImplementedError(f"save_movie is not available for config={self.config}")
+
+        if not HAS_OPENCV:
+            raise ImportError("Please install opencv-python to save cube as movie.")
+
+        if filetype is None:
+            if filename is not None and '.' in filename:
+                filetype = filename.split('.')[-1]
+            else:
+                # default to MPEG-4
+                filetype = "mp4"
+
+        if filetype != "mp4":
+            raise NotImplementedError(f"filetype={filetype} not supported")
+
+        viewer = self.viewer.selected_obj
+        if not isinstance(viewer, BqplotImageView):  # Profile viewer in glue-jupyter cannot do this
+            raise TypeError(f"Movie for {viewer.__class__.__name__} is not supported.")
+        if viewer.shape is None:
+            raise ValueError("Selected viewer has no display shape.")
+
+        if fps is None:
+            fps = float(self.movie_fps)
+        if fps <= 0:
+            raise ValueError("Invalid frame rate, must be positive non-zero value.")
+
+        if filename is None:
+            if self.movie_filename:
+                filename = self.movie_filename
+            else:
+                raise ValueError("Invalid filename.")
+
+        # Make sure file does not end up in weird places in standalone mode.
+        path = os.path.dirname(filename)
+        if path and not os.path.exists(path):
+            raise ValueError(f"Invalid path={path}")
+        elif (not path or path.startswith("..")) and os.environ.get("JDAVIZ_START_DIR", ""):  # noqa: E501 # pragma: no cover
+            filename = os.path.join(os.environ["JDAVIZ_START_DIR"], filename)
+
+        if i_start is None:
+            i_start = int(self.i_start)
+
+        if i_end is None:
+            i_end = int(self.i_end)
+
+        # No wrapping. Forward only.
+        slice_plg = self.app._jdaviz_helper.plugins["Slice"]._obj
+        if i_start < 0:  # pragma: no cover
+            i_start = 0
+        if i_end > slice_plg.max_value:  # pragma: no cover
+            i_end = slice_plg.max_value

```

Comment:
should these cases be caught in UI validation or just done silently?  If there is no validation, the user might expect negative indices to work (or maybe we could support negative indices?).

Reply:
I don't see why someone would want to record a movie backwards (unless they making a horror movie). Anyways, I added validation. I wouldn't worry about negative index support unless we have real users asking for it.